### DISCUSSION TITLE
Problem with left outerjoin, joinedload, limit, and offset

### DISCUSSION THREAD

**Discussion by minyoung90** [sqlalchemy/sqlalchemy#7951] [Usage Questions] [answered]
Hello, I have a problem with joinedload

Here are toy tables,

```python
class User(Base):
    id = Column(BigInteger, primary_key=True, autoincrement=True)
    subscriptions = relationship(
        "Subscription",
        back_populates="subscriber",
        cascade="all, delete",
        passive_deletes=True,
    )

class OTT(Base):
    id = Column(BigInteger, primary_key=True, autoincrement=True)
    readers = relationship(
        "Subscription",
        back_populates="ott",
        cascade="all, delete",
        passive_deletes=True,
    )

class Subscription(Base):
    id = Column(BigInteger, primary_key=True, autoincrement=True)
    user_id = sa.Column(
        sa.ForeignKey(user.id, ondelete="CASCADE"), nullable=False,
    )
    subscriber = sao.relationship(
        User,
        foreign_keys=[user_id],
        back_populates="subscriptions",
        uselist=False,
    )
    ott_id = sa.Column(
        sa.ForeignKey(OTT.id, ondelete="CASCADE"), nullable=False,
    )
    ott = sao.relationship(
        OTT,
        foreign_keys=[ott_id],
        back_populates="readers",
        uselist=False,
    )
```

and example data...

User
| id  |
| --- |
| 1 |
| 2 |
| 3 |

OTT
| id  |
| --- |
| 1 |
| 2 |
| 3 |

Subscription
| id  | user_id | ott_id |
| --- | --- | --- |
| 1 |  1 | 1 |
| 2 | 1 | 2 |
| 3 | 1 | 3 |
| 4 | 2 | 1 |
| 5 | 2 | 2 |
| 6 | 3 | 3 |

So, what I want to do is to get all users who have ott_id == 1 or null (*also pagination).
Expected result is...

| id as User's | id Subscription's | user_id | ott_id |
| --- | --- | --- | --- |
| 1 | 1 | 1 | 1 |
| 2 | 4 | 2 | 1 |
| 3 | 6 | 3 | null |

I used this orm query

```python
select(User)
.options(joinedload(User.subscriptions, innerjoin=False))
.where(
    or_(
        Subscription.ott_id == 1,
        Subscription.ott_id == None,
    )
).limit(20).offset(0)
```

this give me the result...

| id as User's | id Subscription's | user_id | ott_id |
| --- | --- | --- | --- |
| 1 | 1 | 1 | 1 |
| 1 | 2 | 1 | 2 |
| 1 | 3 | 1 | 3 |
| 2 | 4 | 2 | 1 |
| 2 | 5 | 2 | 2 |
| 3 | 6 | 3 | null |

I checked query, there was a subquery with outerjoin filters conditions... but outer of subquery, no conditions applied.

Could you give me a example of orm query for this problem?

**Comment by zzzeek** [comment, marked as answer]
hey there -

dont use joinedload() to create joins that affect query results, use [select.join()](https://docs.sqlalchemy.org/en/14/orm/queryguide.html#joins).   the joinedload() loader option isolates itself from the SQL so that it only performs the task of loading related objects in a collection or attribute without being affected by the rest of the query.   this is discussed at [the Zen of Eager Loading](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#the-zen-of-joined-eager-loading).

**Reply by CaselIT** [reply]
if you use an explicit join, you can use contains_eager to make the orm use the existing join to load the relationship

**Reply by minyoung90** [reply]
@zzzeek @CaselIT  Thanks!!
